### PR TITLE
Fix compiler and linker warning. Drop packing of gyro struct

### PR DESCRIPTION
**Pull request by DzikuVx**
For unknown reason `gyroDev_t` was packed. Removing this peculiarity has a potential of being a breaking change. Maybe someone wanted to save some bytes of RAM, maybe reasoning was different and some extra operations on this struct require gyroDev_t to be packed....

In general, `__attribute__((packed))` can lead to code bloat and slower performance. 

Tested on a workbench with MatekF722 and seems to be working just fine. Tests on more boards would be welcome

### CODE REVIEW

File: src/main/common/typeconversion.c [hunk 1]
```diff
@@ -188,7 +188,7 @@ char *ftoa(float x, char *floatString)
 
     dpLocation = strlen(intString2) - 3;
 
-    strncpy(floatString, intString2, dpLocation);
+    memcpy(floatString, intString2, dpLocation);
```
Comment (gretel) [inline src/main/common/typeconversion.c]: Codacy Issue found: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956)

**Comment by gretel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956&bid=17071774)

**Comment by gretel** [comment]
@DzikuVx thank you! please check on the review

**Comment by Quick-Flash** [comment]
I do like this idea, but is it worth our time? We are planning on merging a lot of the bf code improvements as soon as 0.3.0 is public. I just compiled bf code to see how that would go and it turns out I get 0 compiler warnings. Is this worth our time or is it better that we hold off and just merge what betaflight has done?
image

**Comment by gretel** [comment]
@Quick-Flash this is something that should have been fixed long ago. yes this should be worth our time cause we can not only focus on having new features, regardless of wether a different codebase compiles or not.

**Comment by Quick-Flash** [comment]
> @Quick-Flash this is something that should have been fixed long ago. yes this should be worth our time cause we can not only focus on having new features, regardless of wether a different codebase compiles or not.

Not entirely sure if you get my line of reasoning here. Once we grab code from current BetaFlight it will be easier to grab that code if we are more similar and they have fixed it already. So its kind of like doubling up on the work...

**Comment by gretel** [comment]
yes it is once the work per branch...fork. not sure why abandoning the fork should be the way to go. now that the release has been taken longish we should easily go for this PR?

**Comment by Quick-Flash** [comment]
> yes it is once the work per branch...fork. not sure why abandoning the fork should be the way to go. now that the release is taking longish we should easily go for this PR?

We are not abandoning this fork. Instead we are just going to cherry pick commits from betaflight. Basically we are going to add every commit to get up to bf 4.2 and then drop the ones we don't want and modify a few along the way. This fork is going to have a long life, but perhaps i see your point in making a 0.3.0 branch and dropping this code in there rather than the master and just getting the fix bf did when we eventually get their with all the cherry picking.

**Comment by gretel** [comment]
me is confident we will take lots of time since we get there cause we already do that. so we not taking this pr now?

**Comment by nerdCopter** [comment]
I flew this code on 2 flight controllers. 2 packs each: Helio v1, FoxeerF405.  No noticeable issues.

**Comment by gretel** [comment]
@nerdCopter you should not fly code containing a possible buffer overflow.

**Comment by gretel** [comment]
@DzikuVx hope you are well! 👨‍⚕

**Comment by DzikuVx** [comment]
@gretel yes, thank you, I'm quite fine

**Comment by nerdCopter** [comment]
Flew 2 more packs on FoxeerF405 but with original `strncpy` instead.
All other mods used.  i.e. unpacked gyro struct works.

**Comment by gretel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956&bid=17071774)

File: src/main/common/typeconversion.c (hunk 1 above)
Comment (gretel) [inline src/main/common/typeconversion.c:191]: Codacy Issue found: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956)

**Comment by gretel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956&bid=17071774)

File: src/main/common/typeconversion.c (hunk 1 above)
Comment (gretel) [inline src/main/common/typeconversion.c:191]: Codacy Issue found: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956)

**Comment by DzikuVx** [comment]
@gretel there will be no feedback since the usage of `ftoa`  prevents potential buffer overflow

**Comment by gretel** [comment]
thank you so much for getting back to us. sorry to bother you.

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956&bid=17071774)

File: src/main/common/typeconversion.c (hunk 1 above)
Comment (nerdCopter) [inline src/main/common/typeconversion.c:191]: Codacy Issue found: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956)

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956&bid=17071774)

File: src/main/common/typeconversion.c (hunk 1 above)
Comment (nerdCopter) [inline src/main/common/typeconversion.c:191]: Codacy Issue found: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956)

**Comment by nerdCopter** [comment]
@Quick-Flash @AndreySemjonov @loutwice 
This was included in Build "958-0.3.1-20200613_buildPRs_without_DJI_Hack" for EmuTestBandits.  IF there are no complaints, i don't see why this should not be considered for merging if it reduces any CPU load.  I've flown it several times without discernible issue.

File: src/main/common/typeconversion.c (hunk 1 above)
Comment (nerdCopter) [reply src/main/common/typeconversion.c:191]: for the records, this is non-issue due `ftoa` as Pawel reported.

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956&bid=17071774)

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956&bid=17071774)

**Comment by nerdCopter** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5204956&bid=17071774)

**Pull request merged**